src/sentry/integrations/utils/codecov.py

```diff
@@ -142,40 +142,41 @@ def get_codecov_data(repo: str, service: str, path: str) -> Tuple[LineCoverage |
     return line_coverage, codecov_url
 
 
-def fetch_codecov_data(config: Any) -> Tuple[Optional[Dict[str, Any]], Optional[str]]:
+def fetch_codecov_data(config: Dict[str, Any]) -> Dict[str, Any]:
+    data = {}
+    message = ""
     try:
         repo = config["repository"].name
         service = config["config"]["provider"]["key"]
         path = config["outcome"]["sourcePath"]
 
         lineCoverage, codecovUrl = get_codecov_data(repo, service, path)
         if lineCoverage and codecovUrl:
-            return {
+            data = {
                 "lineCoverage": lineCoverage,
                 "coverageUrl": codecovUrl,
                 "status": status.HTTP_200_OK,
-            }, None
+            }
     except requests.exceptions.HTTPError as error:
         data = {
             "attemptedUrl": error.response.url,
             "status": error.response.status_code,
         }
 
-        message = None
-        if error.response.status_code == status.HTTP_404_NOT_FOUND:
-            message = "Failed to get expected data from Codecov. Continuing execution."
-
-        return data, message
+        # Do not report an error when coverage is not found
+        if error.response.status_code != status.HTTP_404_NOT_FOUND:
+            message = f"Codecov HTTP error: {error.response.status_code}. Continuing execution."
     except requests.Timeout:
         with configure_scope() as scope:
             scope.set_tag("codecov.timeout", True)
             scope.set_tag("codecov.timeout_secs", CODECOV_TIMEOUT)
-        return {
-            "status": status.HTTP_408_REQUEST_TIMEOUT,
-        }, "Codecov request timed out. Continuing execution."
-    except Exception:
-        return {
-            "status": status.HTTP_500_INTERNAL_SERVER_ERROR,
-        }, "Something unexpected happened. Continuing execution."
-
-    return None, None
+            scope.set_tag("codecov.http_code", status.HTTP_408_REQUEST_TIMEOU)
+        data = {"status": status.HTTP_408_REQUEST_TIMEOUT}
+    except Exception as error:
+        data = {"status": status.HTTP_500_INTERNAL_SERVER_ERROR}
+        message = f"{error}. Continuing execution."
```

Comment:
Doing it like this helps us not to loose what was the actual error in the message.